util/unifyfs-stage/src/unifyfs-stage.c

```diff
@@ -0,0 +1,292 @@
+int verbose;
+
+static int debug;
+static int checksum;
+static int mode;
+static char* manifest_file;
+static char* mountpoint = "/unifyfs";
+static char* share_dir;
+
+static unifyfs_stage_t _ctx;
+
+/**
+ * @brief create a status file for unifyfs cli to get niotified with the data
+ * stage result
+ *
+ * @param status 0 indicates success
+ *
+ * @return 0 on success, errno otherwise
+ */
+static int create_status_file(int status)
+{
+    char filename[PATH_MAX] = { 0, };
+    FILE* fp = NULL;
+    const char* msg = status ? "fail" : "success";
+
+    sprintf(filename, "%s/%s", share_dir, UNIFYFS_STAGE_STATUS_FILENAME);
+
+    fp = fopen(filename, "w");
+    if (!fp) {
+        fprintf(stderr, "failed to create %s (%s)\n",
+                        filename, strerror(errno));
+        return errno;
+    }
+
+    fprintf(fp, "%s\n", msg);
+
+    fclose(fp);
+
+    return 0;
+}
+
+static struct option long_opts[] = {
+    { "checksum", 0, 0, 'c' },
+    { "debug", 0, 0, 'd' },
+    { "help", 0, 0, 'h' },
+    { "mountpoint", 1, 0, 'm' },
+    { "parallel", 0, 0, 'p' },
+    { "share-dir", 1, 0, 's' },
+    { "verbose", 0, 0, 'v' },
+    { 0, 0, 0, 0 },
+};
+
+static char* short_opts = "cdhm:ps:v";
+
+static const char* usage_str =
+    "\n"
+    "Usage: %s [OPTION]... <manifest file>\n"
+    "\n"
+    "Transfer files between unifyfs volume and external file system.\n"
+    "The <manifest file> should contain list of files to be transferred,\n"
+    "and each line should be formatted as\n"
+    "\n"
+    "  /source/file/path,/destination/file/path\n"
+    "\n"
+    "Specifying directories is not supported.\n"
+    "\n"
+    "Available options:\n"
+    "\n"
+    "  -c, --checksum           verify md5 checksum for each transfer\n"
+    "  -h, --help               print this usage\n"
+    "  -m, --mountpoint=<mnt>   use <mnt> as unifyfs mountpoint\n"
+    "                           (default: /unifyfs)\n"
+    "  -p, --parallel           transfer each file in parallel\n"
+    "                           (experimental)\n"
+    "  -s, --share-dir=<path>   directory path for creating status file\n"
+    "  -v, --verbose            print noisy outputs\n"
+    "\n"
+    "Without the '-p, --parallel' option, a file is transferred by a single\n"
+    "process. If the '-p, --parallel' option is specified, each file will be\n"
+    "divided by multiple processes and transferred in parallel.\n"
+    "\n";
+
+static char* program;
+
+static void print_usage(void)
+{
+    if (0 == rank) {
+        fprintf(stdout, usage_str, program);
+    }
+}
+
+static inline
+void debug_pause(int rank, const char* fmt, ...)
+{
+    if (rank == 0) {
+        va_list args;
+
+        va_start(args, fmt);
+        vfprintf(stderr, fmt, args);
+        va_end(args);
+
+        fprintf(stderr, " ENTER to continue ... ");
+
+        (void) getchar();
+    }
+
+    MPI_Barrier(MPI_COMM_WORLD);
+
+    /* internal accept() call from mpi may set errno */
+    errno = 0;
+}
```

Comment:
Can we get rid of this?

Reply:
This is really a developer convenience method (used by Hyogi mostly) to make it so that he can attach a debugger at the beginning of the program. It can be useful.

Reply:
@tonyhutter I am mostly using it because it's convenient to attach from gdb. I don't think it's a big problem to keep the code.